### PR TITLE
Implement the itemPathFromRootFind service

### LINKED ISSUE #602: New service to find the path to an item
## Overview 

In case the path to an item is not given or invalid (for instance, if a link is shared between users, and just provided), we try to find a path to the item which is valid for the user.

On `GET /items/:id/path_from_root`

## Input
* items
* as_team_id

## Output
* one path of items from one of the participant's root activities/skills to the input item. All items of the path must be visible (by the participant) and must not include item which requires_explicit_entry and not started, ended, or not allowing submission. 
* If there are multiple solution return one of them. 
* the path is an array of item_ids.
* 'forbidden' if there is no such path

## Validations
* usual as_team_id check

### PR DESCRIPTION
**Pull request by zenovich**
Fixes #602

### CODE REVIEW

File: app/api/items/path_from_root.feature [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      | 3  | 102            | 10           | 2019-05-30 11:00:00 | null                | 2019-05-30 11:00:00      |
+    And the database has the following table 'results':
+      | attempt_id | participant_id | item_id | started_at          | latest_activity_at  | result_propagation_state |
+      | 1          | 102            | 10      | 2019-05-30 11:00:00 | 2019-05-30 11:00:00 | done                     |
+      | 2          | 102            | 10      | 2019-05-30 11:00:00 | 2019-05-30 11:00:00 | done                     |
+      | 2          | 102            | 60      | 2019-05-30 11:00:00 | 2019-05-30 11:00:00 | done                     |
+      | 3          | 102            | 10      | 2019-05-30 11:00:00 | 2019-05-30 11:00:00 | done                     |
+      | 3          | 102            | 60      | 2019-05-30 11:00:00 | 2019-05-30 11:00:00 | done                     |
+
+  Scenario Outline: Find a path for the user's self group
```
Comment (smadbe) [inline app/api/items/path_from_root.feature:60]: I don't get what you mean by "user's self group" here

Comment (zenovich) [reply app/api/items/path_from_root.feature:60]: That means 'no as_team_id parameter'

Comment (smadbe) [reply app/api/items/path_from_root.feature:60]: I don't get how "user's self group" mean "as a user", but ok ;-)

**Review by smadbe — APPROVED** [review]

File: app/api/items/path_from_root.feature (hunk 1 above)
Comment (zenovich) [reply app/api/items/path_from_root.feature]: fixed

**Review by smadbe — APPROVED** [review]
Sorry, I didn't see you re-asked for a review.

**Pull request merged**